appscale/tools/agents/azure_agent.py

```diff
@@ -723,28 +723,41 @@ def cleanup_state(self, parameters):
     network_client = NetworkManagementClient(credentials, subscription_id)
     verbose = parameters[self.PARAM_VERBOSE]
 
-    AppScaleLogger.log("Deleting the Virtual Network, Public IP Address "
-      "and Network Interface created for this deployment.")
+    AppScaleLogger.log("Cleaning up the network configuration created for this "
+                       "deployment ...")
     network_interfaces = network_client.network_interfaces.list(resource_group)
     for interface in network_interfaces:
       result = network_client.network_interfaces.delete(resource_group, interface.name)
       resource_name = 'Network Interface' + ':' + interface.name
       self.sleep_until_delete_operation_done(result, resource_name,
                                              self.MAX_SLEEP_TIME, verbose)
+      AppScaleLogger.verbose("Network Interface {} has been successfully deleted.".
+                             format(interface.name), verbose)
+
+    AppScaleLogger.log("Network Interface(s) have been successfully deleted.")
 
     public_ip_addresses = network_client.public_ip_addresses.list(resource_group)
     for public_ip in public_ip_addresses:
       result = network_client.public_ip_addresses.delete(resource_group, public_ip.name)
       resource_name = 'Public IP Address' + ':' + public_ip.name
       self.sleep_until_delete_operation_done(result, resource_name,
                                              self.MAX_SLEEP_TIME, verbose)
+      AppScaleLogger.verbose("Public IP Address {} has been successfully deleted.".
+                             format(public_ip.name), verbose)
+
+    AppScaleLogger.log("Public IP Address(s) have been successfully deleted.")
 
     virtual_networks = network_client.virtual_networks.list(resource_group)
     for network in virtual_networks:
       result = network_client.virtual_networks.delete(resource_group, network.name)
       resource_name = 'Virtual Network' + ':' + network.name
       self.sleep_until_delete_operation_done(result, resource_name,
                                              self.MAX_SLEEP_TIME, verbose)
+      AppScaleLogger.verbose("Virtual Network {} has been successfully deleted.".
+                             format(network.name), verbose)
+
+    AppScaleLogger.log("Virtual Network(s) have been successfully deleted.")
+    LocalState.cleanup_keyname(parameters[self.PARAM_KEYNAME])
```

Comment:
@obino @tmarballi It used to be in the proper place IMHO: https://github.com/AppScale/appscale-tools/blob/b9c7fd660fc12de575a3af49636ef3496b8dc46a/appscale/tools/appscale_tools.py#L627

but it was removed from there.